drivers/rescontainervbox/main.go

```diff
@@ -0,0 +1,752 @@
+package rescontainervbox
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/antchfx/xmlquery"
+	"github.com/go-ping/ping"
+	"github.com/google/uuid"
+	"github.com/rs/zerolog"
+	"golang.org/x/crypto/ssh"
+	"golang.org/x/crypto/ssh/terminal"
+
+	"github.com/opensvc/om3/core/actionrollback"
+	"github.com/opensvc/om3/core/object"
+	"github.com/opensvc/om3/core/path"
+	"github.com/opensvc/om3/core/provisioned"
+	"github.com/opensvc/om3/core/resource"
+	"github.com/opensvc/om3/core/status"
+	"github.com/opensvc/om3/core/topology"
+	"github.com/opensvc/om3/util/capabilities"
+	"github.com/opensvc/om3/util/command"
+	"github.com/opensvc/om3/util/device"
+	"github.com/opensvc/om3/util/file"
+	"github.com/opensvc/om3/util/sshnode"
+)
+
+const (
+	cpusetDir = "/sys/fs/cgroup/cpuset"
+)
+
+type (
+	T struct {
+		resource.T
+		resource.SCSIPersistentReservation
+		Path     path.T    `json:"path"`
+		ObjectID uuid.UUID `json:"object_id"`
+		Peers    []string  `json:"peers"`
+		DNS      []string  `json:"dns"`
+		Topology topology.T
+
+		Headless       bool `json:"headless"`
+		SCSIReserv     bool `json:"scsireserv"`
+		PromoteRW      bool `json:"promote_rw"`
+		NoPreemptAbort bool `json:"no_preempt_abort"`
+
+		OsvcRootPath string         `json:"osvc_root_path"`
+		GuestOS      string         `json:"guest_os"`
+		Name         string         `json:"name"`
+		Hostname     string         `json:"hostname"`
+		RCmd         []string       `json:"rcmd"`
+		StartTimeout *time.Duration `json:"start_timeout"`
+		StopTimeout  *time.Duration `json:"stop_timeout"`
+
+		cache map[string]interface{}
+	}
+
+	header interface {
+		Head() string
+	}
+	resourceLister interface {
+		Resources() resource.Drivers
+	}
+)
+
+func New() resource.Driver {
+	t := &T{
+		cache: make(map[string]interface{}),
+	}
+	return t
+}
+
+func (t *T) configFile() string {
+	return filepath.Join("/root/VirtualBox VMs", t.Name, t.Name+".vbox")
+}
+
+func (t *T) configFiles() []string {
+	if !t.IsShared() && t.Topology != topology.Failover {
```

Comment:
Single Responsibility Principle (configFiles responsibility is just return the config files)